hydrography-approach/processing_scripts/bridge_statistics/create_bridge_stats.py

```diff
@@ -0,0 +1,212 @@
+import pandas as pd
+import geopandas as gpd
+import os
+from typing import List,Tuple
+import logging
+
+# Set up logging
+logging.basicConfig(level=logging.ERROR, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+def get_gpkg_length(gpkg_file: str) -> int:
+    """
+    Reads a GeoPackage file and returns its length.
+
+    Args:
+        gpkg_file (str): Path to GeoPackage file to read.
+
+    Returns:
+        int: Length of GeoPackage file.
+
+    Raises:
+        FileNotFoundError: If the specified file does not exist.
+        ValueError: If the file is not a valid GeoPackage file.
+    """
+    try:
+        gdf = gpd.read_file(gpkg_file)
+        return len(gdf)
+    except FileNotFoundError as e:
+        logger.error(f"File {gpkg_file} does not exist: {str(e)}", exc_info=True)
+        raise FileNotFoundError(f"File {gpkg_file} does not exist.")
+    except ValueError as e:
+        logger.error(f"{gpkg_file} is not a valid GeoPackage file: {str(e)}", exc_info=True)
+        raise ValueError(f"{gpkg_file} is not a valid GeoPackage file.")
+    except Exception as e:
+        logger.error(f"Unexpected error while reading GeoPackage file {gpkg_file}: {str(e)}", exc_info=True)
+        raise
+    
+
+def get_csv_length(csv_file: str) -> int:
+    """
+    Reads a GeoPackage file and returns its length.
```

Comment:
The docstring for get_csv_length incorrectly states it reads a GeoPackage file. It should specify it reads a CSV file.

Reply:
Corrected it.